telegram/ext/builders.py

```diff
@@ -0,0 +1,1338 @@
+    @check_if_already_set
+    def _set_private_key(self: BuilderType, private_key: bytes) -> BuilderType:
+        if self._bot_was_set:
+            raise self._exception_builder('private_key', 'bot instance')
+        if self._dispatcher_check:
+            raise self._exception_builder('private_key', 'Dispatcher instance')
+        self._private_key = private_key
+        self._private_key_was_set = True
+        return self
+
+    @check_if_already_set
+    def _set_private_key_password(self: BuilderType, private_key_password: bytes) -> BuilderType:
+        if self._bot_was_set:
+            raise self._exception_builder('private_key_password', 'bot instance')
+        if self._dispatcher_check:
+            raise self._exception_builder('private_key_password', 'Dispatcher instance')
+        self._private_key_password = private_key_password
+        self._private_key_password_was_set = True
+        return self
+
+    @check_if_already_set
+    def _set_defaults(self: BuilderType, defaults: 'Defaults') -> BuilderType:
+        if self._bot_was_set:
+            raise self._exception_builder('defaults', 'bot instance')
+        if self._dispatcher_check:
+            raise self._exception_builder('defaults', 'Dispatcher instance')
+        self._defaults = defaults
+        self._defaults_was_set = True
+        return self
+
+    @check_if_already_set
+    def _set_arbitrary_callback_data(
+        self: BuilderType, arbitrary_callback_data: Union[bool, int]
+    ) -> BuilderType:
+        if self._bot_was_set:
+            raise self._exception_builder('arbitrary_callback_data', 'bot instance')
+        if self._dispatcher_check:
+            raise self._exception_builder('arbitrary_callback_data', 'Dispatcher instance')
+        self._arbitrary_callback_data = arbitrary_callback_data
+        self._arbitrary_callback_data_was_set = True
+        return self
+
+    @check_if_already_set
+    def _set_bot(
+        self: '_BaseBuilder[Dispatcher[BT, CCT, UD, CD, BD, JQ, PT], BT, CCT, UD, CD, BD, '
+        'JQ, PT]',
+        bot: InBT,
+    ) -> '_BaseBuilder[Dispatcher[InBT, CCT, UD, CD, BD, JQ, PT], InBT, CCT, UD, CD, BD, JQ, PT]':
+        for attr, error in _BOT_CHECKS:
+            if (
+                getattr(self, f'_{attr}_was_set')
+                if attr != 'dispatcher'
+                else self._dispatcher_check
+            ):
+                raise self._exception_builder('bot', error)
+        self._bot = bot
+        self._bot_was_set = True
+        return self  # type: ignore[return-value]
+
+    @check_if_already_set
+    def _set_update_queue(self: BuilderType, update_queue: Queue) -> BuilderType:
+        if self._dispatcher_check:
+            raise self._exception_builder('update_queue', 'Dispatcher instance')
+        self._update_queue = update_queue
+        self._update_queue_was_set = True
+        return self
+
+    @check_if_already_set
+    def _set_workers(self: BuilderType, workers: int) -> BuilderType:
```

Comment:
In Rust there is `NonZeroU16`. I wonder if there is some equivalent in python (instead of `int`).

Reply:
not AFAIK … the closet you can maybe get is `Literal[1,2,3,4]` which will allow only those numbers and iirc `Literal` is also new in py3.9